userspace/libsinsp/sinsp.cpp

```diff
@@ -1056,6 +1056,52 @@ uint64_t sinsp::max_buf_used()
 	}
 }
 
+void sinsp::get_procs_cpu_from_driver(uint64_t ts)
+{
+	struct timeval tod;
+
+	if(ts <= m_next_flush_time_ns)
+	{
+		return;
+	}
+
+	uint64_t next_full_second = ts - (ts % ONE_SECOND_IN_NS) + ONE_SECOND_IN_NS;
+
+	if(m_next_flush_time_ns == 0)
+	{
+		m_next_flush_time_ns = next_full_second;
+		return;
+	}
+
+	m_next_flush_time_ns = next_full_second;
+
+	gettimeofday(&tod, NULL);
+	uint64_t procrequest_tod = (uint64_t)tod.tv_sec * 1000000000 + tod.tv_usec * 1000;
+
+	if(procrequest_tod - m_last_procrequest_tod <= ONE_SECOND_IN_NS / 2)
+	{
+		return;
+	}
+
+	m_last_procrequest_tod = procrequest_tod;
+
+	m_meinfo.m_pli = scap_get_threadlist(m_h);
+	if(m_meinfo.m_pli == NULL)
+	{
+		throw sinsp_exception(string("scap error: ") + scap_getlasterr(m_h));
+	}
+
+	m_meinfo.m_n_procinfo_evts = m_meinfo.m_pli->n_entries;
+	if(m_meinfo.m_n_procinfo_evts > 0)
+	{
+		m_meinfo.m_cur_procinfo_evt = -1;
```

Comment:
I need to follow this code path a bit more before I actually understand it, because I feel like we're reading off the beginning of the array inside `schedule_next_threadinfo_evt` when we set this to -1. There's probably something I'm missing.

Reply:
This is not new code, so the major concern is whether I broke things while reorganizing it. `void schedule_next_threadinfo_evt(sinsp* _this, void* data)` does look suspect (at a glance we do dereference the -1st element of the array) but I'd say it's for another PR :)

Reply:
Yeah, I just want to make sure I understand what the old code did before I can verify the new code didn't break anything. I guess I can just work off the assumption that this code is either 1) doing something cleverer than I am or 2) has always worked by accident